File: Tools/Notify/src/models/notification_event.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

use crate::models::notification::{EventScope, EventType};

#[derive(Serialize, Deserialize)]
pub struct NotificationEvent {
    #[serde(rename = "type")]
    pub etype: EventType,
    pub scope: EventScope,
    pub payload: Value,
}

impl NotificationEvent {
    pub fn new(etype: EventType, scope: EventScope, payload: Value) -> Self {
        Self {
            etype,
            scope,
            payload,
        }
    }
}

File: Backend/src/main/java/com/anteiku/backend/notification/event/NotificationEvent.java
package com.anteiku.backend.notification.event;

import com.anteiku.backend.notification.payload.NotificationPayload;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter @Setter
public class NotificationEvent {
    EventType type;
    EventScope scope;
    NotificationPayload payload;
}
